distributed/tests/test_worker.py

```diff
@@ -213,6 +213,37 @@ def g(x):
     bb.close_rpc()
     assert not os.path.exists(os.path.join(a.local_dir, eggname))
 
+@gen_cluster(client=True)
+def test_upload_pyz(c, s, a, b):
+    pyzname = 'mytest.pyz'
+    local_file = __file__.replace('test_worker.py', pyzname)
+    assert not os.path.exists(os.path.join(a.local_dir, pyzname))
+    assert not os.path.exists(os.path.join(b.local_dir, pyzname))
+    assert a.local_dir != b.local_dir
+
+    aa = rpc(a.address)
+    bb = rpc(b.address)
+    with open(local_file, 'rb') as f:
+        payload = f.read()
+    yield [aa.upload_file(filename=pyzname, data=payload),
+           bb.upload_file(filename=pyzname, data=payload)]
+
+    assert os.path.exists(os.path.join(a.local_dir, pyzname))
+    assert os.path.exists(os.path.join(b.local_dir, pyzname))
```

Comment:
Thoughts on using `c.upload_file` instead here?  This might be good both to simplify the testing process and also to ensure that everything works at the user-level API.

Reply:
You're right.
I've to admit that my test is inspired from the egg case. Should I change this one too ?

Reply:
actually `c.upload_file` fails because it can't find mytest.pyz
Maybe it's due to the way we create the test cluster, i'm going to investigate this

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Client: not connected>, filename = 'mytest.pyz', raise_on_error = True

    @gen.coroutine
    def _upload_file(self, filename, raise_on_error=True):
>       with open(filename, 'rb') as f:
E       FileNotFoundError: [Errno 2] No such file or directory: 'mytest.pyz'

distributed\client.py:2340: FileNotFoundError`
```

Reply:
You could probably pass it the full path that you create in `local_file`

Reply:
Changing the egg case would be welcome as well.

Reply:
Better late than never!
I made the the appropriate changes. It works fine fo the .pyz case.
However for .egg case, i had to rename the egg from 'mytestegg-1.0.0-py3.4.egg' to 'testegg-1.0.0-py3.4.egg'. In import_file, the module to load is determined using pkg_resource which returns 'mytestegg' instead of 'testegg'.
Not sure if the issue is due to the egg naming or in the way import_file determines the list of modules to load...